tools/azure-devtools/src/azure_devtools/scenario_tests/preparers.py

```diff
@@ -41,9 +41,12 @@ def _prepare_create_resource(self, test_class_instance, **kwargs):
         self.test_class_instance = test_class_instance
 
         if self.live_test or test_class_instance.in_recording:
```

Comment:
This on me since I'm the one who had you put the fix in below, but we can be a little more terse: (My only defense being that I saw it now looking at the fix holistically and not as a one-off >< )

But we can simply make the conditional and leave the core logic all as was prior.  LMK if I'm being dense and this isn't as clean as it seems. (removes the duplicate inner else)
```suggestion
        if (self.live_test or test_class_instance.in_recording) and not self._use_cache:
```